### PR TITLE
[Consensus] Fix testing framework concurrency and improve logging

### PR DESCRIPTION
**Pull request by Olshansk**
## Description

The original motivation for this PR was to fix the `block is nil` on LocalNet.

While testing (using unit tests), debugging and investigating (using LocalNet), several issues were uncovered in the consensus testing framework introduced by #198. The goal of this PR changed to fix those issues and improve the quality of the consensus codebase.

## Issue

Tends to "quality of life" improvements, which are somewhat related to #361.

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [x] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [x] Documentation
- [ ] Other 

### Consensus - Core

- Force consensus to use a "star-like" broadcast instead of "RainTree" broadcast
- Improve logging throughout through the use of emojis and rewording certain statements
- Slightly improve the block verification flow (renaming, minor fixes, etc…) to stabilize LocalNet

### Consensus - Tests

- Rename the `consensus_tests` package to `e2e_tests`
- Internalize configuration related to `fail_on_extra_msgs` from the `Makefile` to the `consensus` module
- Forced all tests to fail if we receive extra unexpected messages and modify tests appropriately
- After #198, we made tests deterministic but there was a hidden bug that modified how the test utility functions because the clock would not move while we were waiting for messages. This prevented logs from streaming, tests from failing, and other issues. Tend to all related changes.

### Consensus - Pacemaker

- Rename `ValidateMessage` to `ShouldHandleMessage` and return a boolean
- Pass a `reason` to `InterruptRoudn`
- Improve readability of some parts of the code

### P2P

- Add a lock to the mempool to avoid parallel messages which has caused the node to crash in the past

### Configs
- Reorder private keys so addresses (retrieved by transforming private keys) to reflect the numbering in LocalNet appropriately. The address for val1, based on config1, will have the lexicographically first address. This makes debugging easier.

## Testing

- [x] `make test_consensus`
- [x] `make develop_test`
- [x] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Required Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested my changes using the available tooling
- [ ] I have updated the corresponding CHANGELOG

### If Applicable Checklist
- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

**Comment by Olshansk** [comment]
@gokutheengineer Just a heads up that we might have some merge conflicts. No action items or need for a review yet.

This is where I'm working through the "block is nil" issue and have been overhauling some of the logs to make things clearer to understand and debug as well. Currently thinking of ways to add better unit tests for it.

**Comment by Olshansk** [comment]
@deblasis I only added @gokutheengineer as a reviewer because I don't think this requires a review from multiple people but just thought you'd appreciate seeing the change here https://github.com/pokt-network/pocket/pull/404/files#diff-351d333685b4f6a9bb38d617a69a200daa5261f20ea31a62a7f1b4123dba9520.

In particular, look at the changes in `waitForEventsInternal`. It was a bug both of us overlooked when adding deterministic testing, but works (and IMO is much clearer) now.

➕ 1️⃣  to the iteration on each other's work

File: consensus/pacemaker.go [hunk 1]
```diff
@@ -185,17 +191,16 @@ func (p *paceMaker) RestartTimer() {
 		select {
 		case <-ctx.Done():
 			if ctx.Err() == context.DeadlineExceeded {
-				p.consensusMod.nodeLog(typesCons.PacemakerTimeout(p.consensusMod.CurrentHeight(), p.consensusMod.step, p.consensusMod.round))
-				p.InterruptRound()
+				p.InterruptRound("timeout")
 			}
 		case <-clock.After(stepTimeout + 30*timePkg.Millisecond): // Adding 30ms to the context timeout to avoid race condition.
```
Comment (gokutheengineer) [inline consensus/pacemaker.go:196]: is there any specific reason to add exactly 30ms, or it is added because it is just enough to avoid race condition?

**Review by gokutheengineer — CHANGES_REQUESTED** [review]

File: consensus/e2e_tests/utils_test.go [hunk 2]
```diff
@@ -98,7 +78,7 @@ func CreateTestConsensusPocketNodes(
 		return pk.Address().String() < pk2.Address().String()
 	})
 	for i, runtimeMgr := range runtimeMgrs {
-		pocketNode := CreateTestConsensusPocketNode(t, &runtimeMgr, testChannel)
+		pocketNode := CreateTestConsensusPocketNode(t, &runtimeMgr, eventsChannel)
 		// TODO(olshansky): Figure this part out.
 		pocketNodes[typesCons.NodeId(i+1)] = pocketNode
```
[the remark this answers is not in the document]
Comment (Olshansk) [reply consensus/e2e_tests/utils_test.go:86]: #434

**Comment by Olshansk** [comment]
@gokutheengineer PTAL when you have a chance

File: consensus/debugging.go [hunk 3]
```diff
@@ -24,12 +27,11 @@ func (m *consensusModule) HandleDebugMessage(debugMessage *messaging.DebugMessag
 }
 
 func (m *consensusModule) GetNodeState() typesCons.ConsensusNodeState {
-	m.m.RLock()
```
[the remark this answers is not in the document]
Comment (Olshansk) [reply consensus/debugging.go:27]: I moved the Lock into `HandleDebugMessage()` above

**Comment by gokutheengineer** [comment]
> @gokutheengineer PTAL when you have a chance

LGTM!

**Review by gokutheengineer — APPROVED** [review]

**Comment by Olshansk** [comment]
@gokutheengineer I'm going to hold of for us to merge in #425  (already approved).

Plan is to rebase this on top of master, retest, and then submit this.

**Pull request merged**